src/ipc/ipc-common/Registry.h

```diff
@@ -112,17 +137,13 @@ class Registry {
         m_content = content;
 
         for (const auto & key : modifiedItems) {
-            if (m_listeners.contains(key)) {
-                for (const auto& a : m_listeners[key]) {
-                    a();
-                }
-            }
+            notifyListeners(key);
         }
 
     }
 
 private:
     QMap<Key, Type> m_content;
-    mutable QMap<Key, QVector<ListenerFunction>> m_listeners;  // "mutable" since adding a listener is not considered as a change
-
+    mutable QMap<Key, QVector<RegistryListenerEntry>> m_listeners;  // "mutable" since adding a listener is not considered as a change
+    QObject* m_owner = nullptr;
```

Comment:
No need to initialize it here with nullptr if you init it in the constructor (there is only one).

Reply:
Bump

Reply:
Not needed with the current code indeed, but it is a precaution to avoid leaving that pointer uninitialized if the constructor happens to be changed later.